### PR TITLE
Update headless-gui action in build (fix screenshots)

### LINKED ISSUE #283: Widget screenshot in `magicgui.md` error
## 🐛 Bug

Widget screenshot in `magicgui.md` (in dev docs: https://napari.org/dev/guides/magicgui.html) looks wrong:

image

Not familiar with `magicgui` so not sure if this is due to our example or the screenshot tool?

### LINKED ISSUE #285: Problem screenshots in "Using magicgui in napari" in dev
## 🐛 Bug

The screenshots in "Using magicgui in napari" in [dev](https://napari.org/dev/guides/magicgui.html#using-magicgui-in-napari) look squished:

image

on [stable](https://napari.org/stable/guides/magicgui.html#using-magicgui-in-napari) the same screenshot:

image

Note this is the worst example, the others on the page look better but still squished compared to stable.

Could it be related to https://github.com/napari/docs/pull/207 @aganders3  ?

### PR DESCRIPTION
**Pull request by aganders3**
# References and relevant issues
~~Hopefully f i x e s~~ Pretty sure now this fixes both #283 and #285 

# Description
This configures the [headless-gui](https://github.com/aganders3/headless-gui) action to not run the tiling window manager it uses by default. My thinking is the tiling behavior in the window manager is causing issues with certain screenshots. See https://github.com/napari/docs/issues/283#issuecomment-1841621067 for more details.

This also updates the action to a new version with a more common default screen size. This change in bit-depth fixes the messed up magicgui screenshot.

Edit: I'm still not _totally_ clear why the stable docs would look okay but I think this is worth trying.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
Fingers crossed!
Stable docs are from July so maybe something changed?
Good news is that I'm pretty sure if the build artifact is fine then the deployed will also be fine, so we can test easily.

Edit2: Oh and @aganders3 could you also--optimistically--edit the deploy action? 
https://github.com/napari/docs/blob/0c7029e5c9674e9d56c5efbec1022fa302661280/.github/workflows/deploy_docs.yml#L56-L65

**Comment by aganders3** [comment]
> Good news is that I'm pretty sure if the build artifact is fine then the deployed will also be fine, so we can test easily.

Nice - hopefully.

Bad news is I think I need to push a change and run it again because I think it needs a non-empty string to avoid the defaults 🤦.

edit: and we need to keep waiting because as I am learning about JavaScript and GH-Actions APIs in real-time 🤦🤦.

**Comment by aganders3** [comment]
Just pointing out a weird traceback here in a recent build - possibly related but let's see how the docs look before I change anything else (for example trying another WM).
https://github.com/napari/docs/actions/runs/7106959550/job/19347465882#step:8:202

**Comment by aganders3** [comment]
Hm, the first screenshot on that page is still messed up, but the rest look better.

I am looking at this build: https://github.com/napari/docs/actions/runs/7106959550

**Comment by psobolewskiPhD** [comment]
Yeah, the size is fixed, but the first widget is messed up. So I guess this fixes #285 only at the moment.

**Comment by aganders3** [comment]
Okay actually I think that worked!

@lucyleeow @psobolewskiPhD can you check the latest build?
https://github.com/napari/docs/actions/runs/7109505974

I converted to draft because this will still need a new release on the action.

**Comment by psobolewskiPhD** [comment]
yes! Looks correct now in the artifact. I don't see issues in other pages!

File: .github/workflows/deploy_docs.yml
Comment (psobolewskiPhD) [inline .github/workflows/deploy_docs.yml]: we need this env var because `make docs` does a pip install.

Comment (lucyleeow) [reply .github/workflows/deploy_docs.yml]: :cry: Its so confusing #291

File: .github/workflows/build_docs.yml
Comment (lucyleeow) [inline .github/workflows/build_docs.yml]: 'setting these empty' refers to skipping 'setup' and 'teardown' ?

Comment (aganders3) [reply .github/workflows/build_docs.yml]: Yeah I'll update this before marking it ready. Initially I set these values empty (`""` or `" "`) but it would still fall back to defaults because I think those are both falsely in JavaScript.

File: .github/workflows/deploy_docs.yml
Comment (aganders3) [reply .github/workflows/deploy_docs.yml]: Ah sorry I think I deleted this on error due to rebasing my local `main`. I'll fix before marking ready to review.

**Comment by psobolewskiPhD** [comment]
@czaki I saw the same problems in the artifact from the 0.4.19 docs PR, so I think this will need to be cherry picked over to that branch.

**Comment by aganders3** [comment]
Okay I'm pretty sure this is good-to-go now using `headless-gui@v2`. Sorry everyone!

**Review by psobolewskiPhD — APPROVED** [review]
I just checked the latest artifact and everything looks good.

File: .github/workflows/build_docs.yml
Comment (psobolewskiPhD) [inline .github/workflows/build_docs.yml]: wonder if we should just stick with pyqt and even consider moving circleCI to pyqt.
pyside (and pyside6 especially) has been the more flaky backend...

Comment (psobolewskiPhD) [reply .github/workflows/build_docs.yml]: Actually, for this PR, I think I'd rather not change the Qt backend, since it's not relevant to the fix and we can make a decision on Qt backends and harmonizing the workflows. Which reminds me, once this is merged, the change needs to also be passed on to:
https://github.com/napari/napari/blob/main/.github/workflows/build_docs.yml
I can take care of that, no problem.

Comment (psobolewskiPhD) [inline .github/workflows/build_docs.yml]:
```suggestion
          python -m pip install "napari/[all]"
```

Comment (aganders3) [reply .github/workflows/build_docs.yml]: I agree and am open to any option here. `all` seemed like a bit much but now that I look it's actually just `pyqt5` + plugin manager. I think it's better to try changing CircleCI to use that.

File: .circleci/config.yml
Comment (psobolewskiPhD) [inline .circleci/config.yml]:
```suggestion
            python -m pip install -e "napari/[pyside,dev]"
```
This is unrelated to this PR, so let's not change the CircleCI workflow which is currently working fine.

**Comment by lucyleeow** [comment]
Should we get another review? Not sure who would be best...

**Comment by lucyleeow** [comment]
Also I question, I see from here: https://github.com/napari/docs/issues/283#issuecomment-1841604776 it seems that the setup in GH and CircleCI is different.

Would trying to mimic Talleys QT setup action to get consistency between CI workflows be a good idea?

**Comment by psobolewskiPhD** [comment]
> Also I question, I see from here: [#283 (comment)](https://github.com/napari/docs/issues/283#issuecomment-1841604776) it seems that the setup in GH and CircleCI is different.
> 
> Would trying to mimic Talleys QT setup action to get consistency between CI workflows be a good idea?

Not sure if the differences are a) required or b) significant. Either way I think changes to CircleCI workflows are out of scope of this PR, but maybe worth noting in an issue?

**Review by Czaki — APPROVED** [review]

**Comment by lucyleeow** [comment]
No problem, it is more from a maintenance point of view - I guess we assume they both do the same thing, and if the docs look fine in CircleCI we assume they would look okay when we deploy but they use *slightly* different workflows? I am no good with Qt and window stuff though. (And I really don't know the ins and outs of our doc CIs)
And yes, further changes are definitely out of scope of this PR.

**Comment by aganders3** [comment]
Thanks for merging @psobolewskiPhD 🚀

Note it did not close the two associated bugs automatically, probably because I edited in the "closing words". 🙁

**Comment by psobolewskiPhD** [comment]
I just wanted to be sure after deployment. Looks good though:
https://napari.org/dev/guides/magicgui.html

**Pull request merged**
